src/composables/useTrackers.ts

```diff
@@ -53,17 +54,30 @@ export function useTrackers(flawUuid: string, affects: Ref<ZodAffectType[]>) {
     })
   );
 
-  const availableUpdateStreams = computed((): UpdateStream[] => moduleComponents.value.flatMap((moduleComponent) =>
-    moduleComponent.streams.map((stream: UpdateStream) => ({
-      ...stream,
-      ps_component: moduleComponent.ps_component,
-      ps_module: moduleComponent.ps_module,
-      affectUuid: moduleComponent.affect.uuid
-    }))
-  ).filter((stream: UpdateStream) => !alreadyFiledTrackers.value.find(
-    (tracker: ZodTrackerTypeWithAffect) => tracker.ps_update_stream === stream.ps_update_stream
+  function isResolutionTrackable(affect: ZodAffectType) {
+    const allowedResolutions: valueof<typeof affectResolutions>[] = [
+      affectResolutions.Delegated,
+      affectResolutions.Empty,
+    ];
+
+    return affect.resolution
+      ? allowedResolutions.includes(affect.resolution)
+      : false;
+  }
```

Comment:
It could, but the additional typing reflects that it is related to the Type definition for affect resolutions. There might be a more elegant way to do this using `in` rather than the method I used above. Thank you, good point

Reply:
[Array.prototype.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) already returns `false` if the item is not found, so the ternary can be removed, also `Set.property.has` is more performant that `Array.prototype.includes` specially if they list grows, but is also noticeable with two elements. [Here's a small benchmark](https://perf.link/#eyJpZCI6Im5qd29xMWJ3cGt6IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoMikua2V5cygpXVxuY29uc3QgZGF0YXNldCA9IG5ldyBTZXQoZGF0YSkiLCJ0ZXN0cyI6W3sibmFtZSI6ImluY2x1ZGVzIHNtYWxsIChzaW11bGF0ZXMgYSAyIGl0ZW0gYXJyYXkpIiwiY29kZSI6ImRhdGEuaW5jbHVkZXMoMSkiLCJydW5zIjpbNTQyMDAwMCwyMTQ3MDAwLDEzMzYwMDAsMTgxMzAwMCwxMjU4MDAwLDQyMzAwMCwyMzc0MDAwLDEwMDA4MDAwLDE3NDMwMDAsMTMwMDAwMCwxMDAwMCwyMzU0MDAwLDkwMDAsMTE4NTAwMCwxMTM0MDAwLDE3ODAwMDAsMjMyNjAwMCwxMjkxMDAwLDkwMDAsOTgwMDAsNzIzNDAwMCwxNzY2MDAwLDY5MDUwMDAsMTY0NjAwMCw5MDAwLDU0MjcwMDAsMTY1NjAwMCwxNzc2MDAwLDkxNTYwMDAsMjM0MDAwLDIwOTMwMDAsMTQwMjAwMCwxNjkzMDAwLDU3MTkwMDAsMTAwMjAwMCw3MjYxMDAwLDQ4NDIwMDAsNTcyMDAwMCwyMzQ1MDAwLDQ1MjgwMDAsNjI2MjAwMCwxMDAwMCwxNjQ2MDAwLDQxNzkwMDAsNDAxNjAwMCw0NTQ2MDAwLDgyMTQwMDAsMTAwNDEwMDAsMTAwMDAsNDAwMDAwLDQzMzEwMDAsMTMzNzAwMDAsMTUyMDAwMCwxMzM3ODAwMCw2MjA4MDAwLDMxNjAwMCw4OTE2MDAwLDEwMjQ3MDAwLDEwMjA3MDAwLDE4MDEwMDAsNjM3MDAwMCw2MTMwMDAsMjEzNTAwMCw1NTA1MDAwLDEzOTQwMDAsOTAwMCw1OTgzMDAwLDk4MDAwLDM3MTUwMDAsMzYxMDAwMCwxMjAwMCw5NDg5MDAwLDYyMDAwLDIxMTAwMCw5ODAwMCwzNDY1MDAwLDMzODcwMDAsNjIxMDAwMCw5MDAwLDExMDI3MDAwLDYxNzEwMDAsOTAwMCwxMTIzMTAwMCw2MDMyMDAwLDExMTE2MDAwLDE2NDYwMDAsMTYxNDAwMCwyNTA1MDAwLDY5NzkwMDAsOTE0MDAwLDEwODQxMDAwLDI5MjkwMDAsODY1MDAwLDEwOTUwMDAwLDk4MDAwLDc1MjIwMDAsNDkwODAwMCwyODYyMDAwLDQyNTMwMDAsOTAwMF0sIm9wcyI6Mzc2OTA2MH0seyJuYW1lIjoiaW5jbHVkZXMgYmlnICh0byBzZWUgcGVyZm9ybWFuY2UpIiwiY29kZSI6ImRhdGFzZXQuaGFzKDEpIiwicnVucyI6WzEwMTgwMDAsNDc1MDAwLDQ2MDcwMDAsMzg4NDAwMCwxMjIwMDAwLDk3ODcwMDAsMjg2NTAwMCwxMjg4NzAwMCwxNjQ2MDAwLDE0MjgwMDAsOTgwMDAsMzU2NzAwMCwzNzk0MDAwLDAsOTgwMDAsMjMwMzAwMCw5NTQwMDAsNDg0OTAwMCw0MDY1MDAwLDM5MjcwMDAsMjA0NDgwMDAsOTAwMCw3MzQ3MDAwLDE4MDAwMDAsOTgwMDAsNTkwMjAwMCwxMjM4NzAwMCw5MDAwLDQwMzAwMCwxMDk5NjAwMCwyMTc0MDAwLDE1MjEwMDAsMTkzMDAwMCw2MTQ5MDAwLDE1MTE1MDAwLDI1NTk3MDAwLDE5NzMwMDAsNjE5MjAwMCw0NTIzMDAwLDkwMDAsMTE1MDAwLDEwMDAwLDY1MjkwMDAsMzg4MjAwMCwxMzc2MzAwMCw5Mjk3MDAwLDEzNjkwMDAsMzUyMTAwMCw0NjM2MDAwLDQ5MjcwMDAsNDMzNTAwMCwxMjA3ODAwMCw2Mzg1MDAwLDkxMDAwLDE1NjgwMDAwLDI5NTYwMDAsOTYzMTAwMCw3MDkwMDAsMTY0NjAwMCw2MjAwMCwzNTUyMDAwLDQ3OTAwMDAsMjAzMjAwMCwxMjM1MTAwMCw5MDAwLDg0OTcwMDAsMzAzNzAwMCw5ODAwMCw0NzMwMDAsMzg1MjAwMCwxMDAwMCwzMTY0MDAwLDIxNTgwMDAsMjA1NzAwMCw5ODAwMCwzODE2MDAwLDExNzg0MDAwLDQ4NjAwMCw5MDAwLDgzNjEwMDAsMTAwMDAsMTAwMDAsMTQyNjAwMCwxMTUyODAwMCw2MDk1MDAwLDE4MTEwMDAsNzQ1MDAwLDk5MTkwMDAsNzk5MzAwMCwzMTQ5MDAwLDEzNjE2MDAwLDMyMjIwMDAsMTk5MDAwLDI0MDgwMDAwLDc3NDAwMCw5MjE4MDAwLDI0NzUwMDAsMzkyNTAwMCwxMTE5MDAwLDE2NDYwMDBdLCJvcHMiOjQ2MzI3MDB9XSwidXBkYXRlZCI6IjIwMjQtMDgtMjBUMDc6NTM6MDkuNzY3WiJ9) 

```suggestion
  function isResolutionTrackable(affect: ZodAffectType) {
    const allowedResolutions = new Set<valueof<typeof affectResolutions>>([
      affectResolutions.Delegated,
      affectResolutions.Empty,
   ]);

    return allowedResolutions.has(affect.resolution);
  }
```

This could be a **type guard** by setting the return type to `function isResolutionTrackable(affect: Affect): affect is (Omit<Affect, 'resolution'> & {resolution: "DELEGATED" | ""})`, but because we are iterating through `modules[].affect` instead of `affects[]` typescript does not infer the types correctly.
[Here's a playground using the guard](https://www.typescriptlang.org/play/?ts=5.2.2#code/MYewdgzgLgBAhgMwQU2FASsiIA2BXKAS3AhgF4YBvAKBhgBFkdkBzOKZAEwC4YByegFEAMoIDiAQQAqg+nwA0tGIIC2ABygBPXnwVLGKAE46hAMUHo9dAPIgIEHdesBlZ32oBfeKVCQo1ai01ZBgANzh8ZBAEAB4pAD5yGCkAbQBrZE1o5IBdAKCQiSRUWAoaOjw8Qh4YMDwVACNkQyVDLFwCYjBecMjomILsxBQ0TGx8IhJ4z3zNYJgAWRBOPGYk8phsFWReaENCMBYlYZLeIpH-DwDqBDwwNC6YQggxjsmwKUM4YDS4BuYABQnNBnYpoACUvGBsGeMAB1hUhCgMXOJXk-Da4064D4iQAZFRMW8urwAERCUSSGT0UkwAA+MFJpI84KoSjovmg8BwOBAAHcuK8Jl1SBQwMg+TBnMhkb08FFYoMEPAwRh2sKpvEASl2XQVRchdjIAA6RjMNgcTiKPV66GG94QY2qDSaa10HLggDcARtbSgeEMYG5vIFnHtIuNAAs4BAgarjUSNWAvTMApzYNCHDBUWgUjkkilKJVqtwAIzyRNG7ikpyuZnootVHgAJnRlfe1Yp4mksmZeWomeNCEIOA4hgBz3D4E+31+-2Q4ONKjgajjF3IiQ2nNwyGNvJYa5KCfVRvIZAo5ME5nQpNZAHo71KDsAQgL4G0nmO4AcWDAoJGQkzdECggYB9g0GBgDgIMDiMP8AL-OYsHgMBOBgNIwH5Uh-1hWFMNgNQ7AgQh508cE0xIWAVGWVYsF4JYVmYPMC0oLYdj4DJNAUfVTioYseHLdsSRrFxnGZDwGzY7gOMyZtuOhXhGxLVshPATsRG7alxP7ajGKwIcRzHAEAUoHi0BgFleEwf1AykJCBiQ7JJxPd4Zx+P5mHiMhEmcrFXK+dz50PCFF2XVddNojc2Q5Egdz3EADwi5hjWhY8-Mec8LzMCxbxgB8YAACWaZB0SaKC8AgV9AI-JFmnYH94JCNQ4DaMBYBABoACs0UQ4JQPA2BOBAZD8KeMA4M4ZBkDUGBkEwvAWEjP8QEgkB1Bwb8gwBZBQjmmAGk0GAKoa-8QmQAAPNQcEIYAkRgayAw+JDyJZIA)